### DISCUSSION TITLE
Making gathering more engaging/fun

### DISCUSSION THREAD

**Discussion by jasper-one** [EveronLife/EveronLife#166] [Ideas]
Players sometimes spend hundreds of hours gathering in Life Mods, one of my wishes for Everon Life is that we make it more engaging. Starting small with the ideas here and in combination with great animations as discussed in #143 and unique gathering types like #74 I think we can make the repetitive task much more engaging then it has ever been. Inspired by other RPGs and Minecraft, the first relatively simple ideas I propose adding are: 

**Rare Drops**
The ability for each gathers type to drop additional items on a percentage basis. This adds some variation to the gathering tasks and will help to make them slightly less repetitive and more fun/rewarding. 

Examples:
- Gravel, Sand, and Concrete could drop Ore Dusts that could be forged into Ingots and sold. 
- Copper, Iron and Gold mining could drop different gems

**Tool Efficiency**
The ability for players to choose from a number of different tools that have different purposes. Players could purchase progressively more expensive and efficient or specialized tools like better Pickaxe, Shovels, etc. Players could also get equipment that specializes on getting the rare drops rather than the core resource. 

I feel like this is a tried and true method based on the Minecraft tool system that I think works great and gives players a good sense of progression and variation on each run that they are trying a new faster, better tool. 

Examples:
- Different shovels, and picks for different efficiency
- Drills for mining that return fewer ores but more gems. 

**Adding Rare Drops and 3 different Shovels as a POC for gathering Sand, Gravel and Cement**

I propose implementing the following 3 shovels as additional tools much like #88 
<img width="172" alt="image" src="https://user-images.githubusercontent.com/106525617/190612820-0bbdd4fb-6225-42a6-a3db-470d6d9c906c.png"><img width="122" alt="image" src="https://user-images.githubusercontent.com/106525617/190612885-5b2e5e07-af56-43e7-918f-211093bbb352.png"><img width="118" alt="image" src="https://user-images.githubusercontent.com/106525617/190612643-c832106c-d6c6-414d-872b-82d200071fcd.png">

Using the above shovels and also creating the following rare drops:
- Copper Dust
- Iron Dust
- Gold Dust 

Rare drops & tool efficiency could then be implemented using the above to update:

- Sand Gathering (with Copper Dust rare drops)
- Gravel Gathering (with Iron Dust rare drops)
- Concrete Gathering (with Gold Dust rare drops)

**Technical Implementation**

Technically, I guess. This would roughly involve adding multiple outputs to the gathering script, each with a default drop percentage. Maybe something like 0.2 for %20 odds of dropping and 2 for 2 items always dropping. There would then also be an array of tools that can (gather requirement tool) where each tool could override the drop percentage/quantities and default gathering time.  

If we agree this is a good idea, I would be happy to have a go at implementing it. I look forward to everyone's thoughts.

**Comment by hiimk3v** [comment]
Maybe during refining there could be a chance to get a higher quality version of that ingot that can be sold for more or used for higher tier crafts.

**Comment by thebonbon** [comment]
Apart from the rare drops, tools and stuff the base gathering system should be engaging and fun (not the hold F for 10sec to gather).

Something more interactive like this or the tree cutting would make it way more engaging:
![appleTree](https://user-images.githubusercontent.com/45827986/190849328-e5680b8e-811d-43ce-9dce-5eaa890a1920.gif)
